tools/restatectl/src/commands/node/disable_node_checker.rs

```diff
@@ -77,6 +107,69 @@ impl<'a, 'b> DisableNodeChecker<'a, 'b> {
         Ok(())
     }
 
+    /// Checks whether it is safe to disable the worker. A worker is safe to disable if it is not
+    /// used in any partition replica set, and it cannot be added to any future replica sets.
+    async fn safe_to_disable_worker(
+        &self,
+        node_id: PlainNodeId,
+        node_config: &NodeConfig,
+    ) -> Result<(), DisableWorkerError> {
+        match node_config.worker_config.worker_state {
+            WorkerState::Active => {
+                return Err(DisableWorkerError::Active);
+            }
```

Comment:
technically, the safety depends on the replication property or partitions located on this worker node. So, this could be a stricter than needed. That said, best to get a feel on how it feels in practice before we make it more relaxed. So, +1 to keeping it as is for the first iteration.

Reply:
👍